File: frontend/src/types/response/invoice.ts
import { ItemResponse } from "./item";
import { TagResponse } from "./tag";

export interface InvoiceItemResponse {
  id: number;
  item_id: number;
  item: ItemResponse;
  quantity: number;
  unit_price: number;
  total_price: number;
}

export interface InvoiceDetailResponse {
  id: number;
  grand_price: number;
  items: InvoiceItemResponse[];
  tags: TagResponse[];
  created_at: string;
  updated_at: string;
}

export interface InvoiceListItem {
  id: number;
  grand_price: number;
  tags: TagResponse[];
  totalItem: number;
  created_at: string;
  updated_at: string;
}

export interface InvoicePaginationMeta {
  totalData: number;
  page: number;
  limit: number;
  totalPage: number;
}

export interface InvoicePaginationResponse {
  data: InvoiceListItem[];
  meta: InvoicePaginationMeta;
}

File: backend/model/response/invoice.go
package response

import (
	"time"

	"github.com/google/uuid"
)

type InvoiceItemResponse struct {
	ID         uuid.UUID    `json:"id"`
	ItemID     uuid.UUID    `json:"item_id"`
	Item       ItemResponse `json:"item"`
	Quantity   int          `json:"quantity"`
	UnitPrice  float64      `json:"unit_price"`
	TotalPrice float64      `json:"total_price"`
}

type InvoiceResponse struct {
	ID         uuid.UUID `json:"id"`
	GrandPrice float64   `json:"grand_price"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type InvoiceDetailResponse struct {
	ID         uuid.UUID             `json:"id"`
	GrandPrice float64               `json:"grand_price"`
	Items      []InvoiceItemResponse `json:"items"`
	Tags       []TagResponse         `json:"tags"`
	CreatedAt  time.Time             `json:"created_at"`
	UpdatedAt  time.Time             `json:"updated_at"`
}

type InvoiceListItem struct {
	ID         uuid.UUID     `json:"id"`
	GrandPrice float64       `json:"grand_price"`
	Tags       []TagResponse `json:"tags"`
	TotalItem  int           `json:"totalItem"`
	CreatedAt  time.Time     `json:"created_at"`
	UpdatedAt  time.Time     `json:"updated_at"`
}

type InvoicePaginationMeta struct {
	TotalData int `json:"totalData"`
	Page      int `json:"page"`
	Limit     int `json:"limit"`
	TotalPage int `json:"totalPage"`
}

type InvoicePaginationResponse struct {
	Data []InvoiceListItem     `json:"data"`
	Meta InvoicePaginationMeta `json:"meta"`
}
